staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -281,13 +284,36 @@ var CommonNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate
 	fp := sha256.Sum256(chain[0].Raw)
 	id := "X509SHA256=" + hex.EncodeToString(fp[:])
 
+	uid := ""
+	if utilfeature.DefaultFeatureGate.Enabled(features.AllowParsingUserUIDFromCertAuth) {
+		uid = parseUIDFromCert(chain[0])
+	}
+
 	return &authenticator.Response{
 		User: &user.DefaultInfo{
 			Name:   chain[0].Subject.CommonName,
+			UID:    uid,
 			Groups: chain[0].Subject.Organization,
 			Extra: map[string][]string{
 				user.CredentialIDKey: {id},
 			},
 		},
 	}, true, nil
 })
+
+// Using CNCF PEN for Kubernetes:
+// https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/4317-pod-certificates#x509-podidentity-extension
+var uidOID = asn1.ObjectIdentifier([]int{1, 3, 6, 1, 4, 1, 57683, 2})
+
+// Go's crypto/x509 library doesn't explicitly serialize UID to a named field, so
+// we read it off the asn1 array.
```

Comment:
This comment doesn't really make sense when using a PEN.

Reply:
Thanks for the catch - I removed it 👍